daemon/stop.go

```diff
@@ -78,12 +78,16 @@ func (daemon *Daemon) containerStop(container *containerpkg.Container, seconds i
 		logrus.Infof("Container %v failed to exit within %d seconds of signal %d - using the force", container.ID, seconds, stopSignal)
 		// 3. If it doesn't, then send SIGKILL
 		if err := daemon.Kill(container); err != nil {
-			// Wait without a timeout, ignore result.
-			<-container.Wait(context.Background(), containerpkg.WaitConditionNotRunning)
-			logrus.Warn(err) // Don't return error because we only care that container is stopped, not what function stopped it
+			logrus.Errorf("Container %v failed to SIGKILL container: %s", container.ID, err)
 		}
 	}
 
+	ctx2, cancel2 := context.WithTimeout(context.Background(), time.Duration(seconds)*time.Second)
+	defer cancel2()
+	if status := <-container.Wait(ctx2, containerpkg.WaitConditionNotRunning); status.Err() != nil {
```

Comment:
If we're still waiting, we probably need to move this back inside of `if err := daemon.Kill(container); err != nil {`

Reply:
yes i agree, i think that is more correct